rpc/types_test.go

```diff
@@ -153,3 +153,33 @@ func TestBlockNumberOrHash_WithNumber_MarshalAndUnmarshal(t *testing.T) {
 		})
 	}
 }
+
+func TestBlockNumberOrHash_WithNumber_StringAndUnmarshal(t *testing.T) {
+	tests := []struct {
+		name  string
+		value BlockNumberOrHash
+	}{
+		{"max", BlockNumberOrHashWithNumber(math.MaxInt64)},
+		{"pending", BlockNumberOrHashWithNumber(PendingBlockNumber)},
+		{"latest", BlockNumberOrHashWithNumber(LatestBlockNumber)},
+		{"earliest", BlockNumberOrHashWithNumber(EarliestBlockNumber)},
+		{"0x20", BlockNumberOrHashWithNumber(32)},
+		{"hash", BlockNumberOrHashWithHash(common.Hash{0xaa}, false)},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			bnh := test.value
+			// Wrap the string value in quotes to make it a JSON string.
+			marshalled := []byte("\"" + bnh.String() + "\"")
+			var unmarshalled BlockNumberOrHash
```

Comment:
```suggestion
			var have BlockNumberOrHash
```
we usually try to wind up with a `have` and a `want`, and the test later compares if have eq want.